### PR TITLE
[proposal] Extract styles from depject

### PR DESCRIPTION
**Pull request by mmckegg**
I've been having some ideas about styling spawned from [recent discussion](https://github.com/ssbc/patchbay/issues/86). Rather than ramble on about them, I thought I'd just go ahead a try it!

I've **extracted all of the styling out of depject** (moving it to styles/*) and made it the responsibility of the client to apply it. So styles would be applied just like they used to before the mcss refactor, except now we can have **multiple style files** and they can be **either css or mcss** (oh and the wonderful new dom structure). 

A nice side effect is we can ditch all of that boilerplate `gives: "mcss"` and `readFileSync` repetition in every module. I can see no downside since the mcss wasn't really being applied modularly anyway.

In this branch, the styles are now applied directly in [index.js](https://github.com/ssbc/patchbay/blob/extract-styles-from-depject/index.js#L14). If you require the modules manually, they'll have no styles by default (except for the inline ones).

I know this is a bit of a ...uh.. _sore_ spot right now. But thoughts? 

/cc @ahdinosaur @dominictarr @evbogue @mixmix

### CODE REVIEW

**Comment by mmckegg** [comment]
BTW, while this is rather relevant to recent "discussions", this is mostly coming from the point of view of merging the latest `patchbay` into `patchwork-next`. 

I would much rather the styles being seperate than tangled up in the depject logic.

**Comment by mmckegg** [comment]
**One more thing:** once patchbay upgrades to use the latest version of `micro-css` (v1.0.0), all of the classNames will become easily stylable from standard css. 

Also it will also be much easier to use mcss to style things that _weren't written for mcss_ since classes are no longer prefixed.

**Comment by ahdinosaur** [comment]
hmm. personally, i'm most interested in finding a way to do style modules using `depject`, since i like system architectures where patterns are consistently applied, and i feel like pulling styles out of depject (and especially into a new top-level folder) is a new edge case (inconsistency) to how we separate functionality into modules. but this is just my own opinion about software architecture.

>  I can see no downside since the mcss wasn't really being applied modularly anyway.

i agree that the current way we're doing style modules is not serving us, keen to explore alternatives. i'm especially interested in the second proposal in https://github.com/ssbc/patchbay/issues/86#issuecomment-278482264, would appreciate feedback.

this all being said, i'm not against merging this, since we can always iterate on modular styles in the future. :heart:

File: styles/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  if (/\.mcss$/i.test(file)) {
+    result += fs.readFileSync(path.resolve(__dirname, file), 'utf8') + '\n'
+  }
+
+  if (/\.css$/i.test(file)) {
+    additional += fs.readFileSync(path.resolve(__dirname, file), 'utf8') + '\n'
+  }
+})
+
+module.exports = compile(result) + additional
```
Comment (mixmix) [inline styles/index.js]: oooo, this is nice!

**Comment by mixmix** [comment]
I'm a fan of Mikey's second proposal - the style modules which has a bult in reduce which can over-ride.
I think I'm attracted to the idea that if I make some widget and you like it, you can grab it, import it, and have some sensible defaults (like some flex + padding) pulled into your app, and if you want to make it purple you can just extend, or if you want to over-ride you can wreck the lot.

I think where patchbay is currently at might not be the sensible minimum. Would you be interested in pairing on taking the styling I wrote and pulling out the sugary opinion bits and seeing where we could put those. I imagine this might become the first 'theme' and then your preferences would become the second one.

**Comment by mmckegg** [comment]
> I think I'm attracted to the idea that if I make some widget and you like it, you can grab it, import it, and have some sensible defaults (like some flex + padding) pulled into your app, and if you want to make it purple you can just extend, or if you want to over-ride you can wreck the lot.

Yeah, I think this is the right approach. 

I've spent the evening working on porting `patchwork-next` to the latest `patchbay` and I've found that most of the styles are okay (and can just be added to), but that I want to **completely override** the `Message` style. Turns out that this is quite difficult to do with my proposal and **a lot easier with the way it is currently**. That's why it's good to try these things! 😆

**Comment by mmckegg** [comment]
@mixmix and yes definitely interested in collaborating on extracting out the sugary bits in the styling. I've also come across a few DOM issues when I wanted to style the messages. We can talk about those too.

**Comment by mixmix** [comment]
closed - thiis has been superseded

**Pull request closed without merging**